lib/controllers/index.ts

```diff
@@ -327,3 +328,43 @@ export function controllerNFTValue(
   );
   return value;
 }
+
+export async function computeNewProjectedAPR(
+  newMark: number,
+  target: number,
+  secondsHeld: number,
+  token: SupportedToken,
+): Promise<number> {
+  const controller = jsonRpcControllerContract(
+    configs[token].controllerAddress,
+    configs[token].jsonRpcProvider,
+    token,
+  );
+
+  const targetMarkRatioMax = parseFloat(
+    ethers.utils.formatEther(await controller.targetMarkRatioMax()),
+  );
+  const targetMarkRatioMin = parseFloat(
+    ethers.utils.formatEther(await controller.targetMarkRatioMin()),
```

Comment:
we should consider if we want these in the subgraph, in the spirit performance and reducing RPC calls